src/e3/encoding/vex.py

```diff
@@ -0,0 +1,902 @@
+
+    def __bool__(self) -> bool:
+        """Check if this justification is defined.
+
+        :return: **False** only if justification is set to
+            :attr:`Justification.NO_JUSTIFICATION`, else **True** is returned.
+        """  # noqa RST304
+        return self != Justification.NO_JUSTIFICATION
+
+    @classmethod
+    def from_value(
+        cls,
+        value: str | Justification | None,
+        default: Justification | str = NO_JUSTIFICATION,
+    ) -> Justification:
+        """Create a justification from a given *value*.
+
+        :return: A justification set according to *value* and *default*.
+
+        :raise: :exc:`python:ValueError` If *value* is not one of the possible
+            values of this enumerate, or if *default* has an invalid value and
+            *value* is **None**.
+        """  # noqa RST304
+        if isinstance(value, str) and value:
+            return cls(value)
+        elif isinstance(value, cls):
+            return value
+        elif default:
+            # The default value is either a Justification object or a string,
+            # better return a Justification.
+            return cls.from_value(default)
+        else:
+            raise ValueError(f"Invalid default value {default}")
+
+
+class Metadata(JsonData):
+    """VEX metadata.
+
+    To the greatest extent possible, VEX metadata is defined and maintained at
+    the VEX document level.
+
+    When appropriate and necessary, VEX metadata is defined at the VEX statement
+    level.
+
+    VEX document metadata **MAY** be synthesized or derived from VEX statement
+    metadata; for example, *doc_time_last_updated* **MUST** be at least as
+    recent as the newest *statement_time_last_updated*.
+
+    VEX document metadata **MUST** accurately apply to all contained VEX
+    statements.
+
+    Must include: VEX Format Identifier, Identifier string for the VEX
+    document, Author, Author role, Timestamp.
+
+    :ivar str author: The author of the VEX document. The *author* is
+        responsible for the content of the VEX document. A VEX document **MUST**
+        identify the author. To describe tools or other mechanisms used to
+        generate VEX content, consider *tooling*.
+    :ivar AuthorRole author_role: The role of the other of this document. See
+        :class:`AuthorRole`.
+    :ivar str tooling: Document tooling **MAY** specify tools or automated
+        mechanisms that generate VEX documents, VEX statements, or other VEX
+        information. Contrast with *author*.
+    :ivar datetime created_on: The time this VEX document was created at.
+    :ivar datetime last_updated_on: The time this VEX document was updated.
+    :ivar int version: The version of a VEX document.
+    :ivar str spec_version: The VEX requirements version used by the parent
+        document. By default, it is set to :attr:`SPEC_VERSION`.
+    """  # noqa RST304
+
+    AUTHOR: str = "AdaCore"
+    """Default document author."""
```

Comment:
Don't use triple quote here and for documentation of internal constant

Reply:
Done